src/jdk.crypto.mscapi/windows/native/libsunmscapi/security.cpp

```diff
@@ -724,19 +689,142 @@ JNIEXPORT void JNICALL Java_sun_security_mscapi_CKeyStore_loadKeysOrCertificateC
         }
     }
     __finally
+    {
```

Comment:
Why is this finally block required now and not prior to adding this feature?

Reply:
Also do we need to clear down the other two variables declared at the top of this function?

Reply:
the next PR fixed a copy+paste error, in this PR it looks like there's a new try block but there is not,  remember the original loadCertficates... method has been changed to internal and takes in the keystore, and the ttwo new methods handle the original opening of keystore using name and location, and the new mechanism of loading from memory